test/e2e/retry_test.go

```diff
@@ -48,6 +49,36 @@ spec:
 		})
 }
 
+func (s *RetryTestSuite) TestRetryBackoff() {
+	s.Given().
+		Workflow(`
+metadata:
+  generateName: test-backoff-strategy-
+spec:
+  entrypoint: main
+  templates:
+    - name: main
+      retryStrategy:
+        limit: 10
```

Comment:
I should've pointed this out last time, since the error is with the intstring conversion maybe we should test both `'10'` and `10`

Reply:
Will add a separate workflow to test this, good idea.